Indicators/DeMarkerIndicator.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// In the DeMarker strategy, for some period of size N, set:
+    /// <para>
+    /// DeMax = High - Previous High, and 
+    /// DeMin = Previous Low - Low
+    /// </para>
+    /// where, in the prior, if either term is less than zero (DeMax or DeMin), set it to zero.
+    /// We can now define the indicator itself, DeM, as:
+    ///<para>
+    /// DeM = MA(DeMax)/(MA(DeMax)+MA(DeMin))
+    ///</para>
+    /// where MA denotes a Moving Average of period N.
+    /// 
+    /// https://www.investopedia.com/terms/d/demarkerindicator.asp
+    /// </summary>
+    public class DeMarkerIndicator : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private decimal LastHigh;
+        private decimal LastLow;
+        private decimal DeMin;
+        private decimal DeMax;
+        private IndicatorBase<IndicatorDataPoint> _MAmax;
+        private IndicatorBase<IndicatorDataPoint> _MAmin;
+        private int _period;
+
+
+        /// <summary>
+        /// Initializes a new instance of the DeMarkerIndicator class with the specified name and period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of the  DeMarker Indicator</param>
+        /// <param name="type">The type of moving average to use in calculations</param>
+        public DeMarkerIndicator(string name, int period, MovingAverageType type = MovingAverageType.Simple )
+            : base(name)
+        {
+            LastHigh = 0m;
+            LastLow = 0m;
+            _period = period;
```

Comment:
We could convert `WarmUpPeriod` to auto-property:
```csharp
WarmUpPeriod = period;
```
and remove `_period` class variable.